scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala

```diff
@@ -193,6 +194,108 @@ class PairSCollectionFunctions[K, V](val self: SCollection[(K, V)])
     MultiJoin.left(that, _).mapValues(kv => (kv._2, kv._1))
   }
 
+  /* Hash operations */
+
+  /**
+   * Perform an inner join by replicating `that` to all workers. The right side should be tiny and
+   * fit in memory.
+   *
+   * @group join
+   */
+  def hashJoin[W: ClassTag](that: SCollection[(K, W)])
+  : SCollection[(K, (V, W))] = self.transform { in =>
+    val side = that.asMultiMapSideInput
+    in.withSideInputs(side).flatMap[(K, (V, W))] { (kv, s) =>
+      s(side).getOrElse(kv._1, Iterable()).toSeq.map(w => (kv._1, (kv._2, w)))
+    }.toSCollection
+  }
+
+  /**
+   * Perform a left outer join by replicating `that` to all workers. The right side should be tiny
+   * and fit in memory.
+   *
+   * @group join
+   */
+  def hashLeftJoin[W: ClassTag](that: SCollection[(K, W)])
+  : SCollection[(K, (V, Option[W]))] = self.transform { in =>
+    val side = that.asMultiMapSideInput
+    in.withSideInputs(side).flatMap[(K, (V, Option[W]))] { (kv, s) =>
+      val (k, v) = kv
+      val m = s(side)
+      if (m.contains(k)) m(k).map(w => (k, (v, Some(w)))) else Seq((k, (v, None)))
+    }.toSCollection
+  }
+
+  /* Skew operations */
+
+  /**
+   * N to 1 skewproof flavor of [[PairSCollectionFunctions.join()]].
```

Comment:
Do we really need N1 in the name? Maybe hard to explain in layman's them. `skewedJoin` sounds clear enough?